state/aws/dynamodb/dynamodb.go

```diff
@@ -393,6 +393,61 @@ func (d *StateStore) parseTTL(req *state.SetRequest) (*int64, error) {
 	return nil, nil
 }
 
+// Multi performs a transactional operation. succeeds only if all operations succeed, and fails if one or more operations fail.
+func (d *StateStore) Multi(ctx context.Context, request *state.TransactionalStateRequest) error {
+	twinput := &dynamodb.TransactWriteItemsInput{
+		TransactItems: []*dynamodb.TransactWriteItem{},
+	}
+
+	// Dedup ops where the last operation with a matching Key takes precedence
+	txs := map[string]int{}
+	for i, o := range request.Operations {
+		txs[o.GetKey()] = i
+	}
+
+	for i, o := range request.Operations {
+		// skip operations removed in simulated set
+		if txs[o.GetKey()] != i {
+			continue
+		}
+
+		twi := &dynamodb.TransactWriteItem{}
+		switch req := o.(type) {
+		case state.SetRequest:
+			value, err := d.marshalToString(req.Value)
+			if err != nil {
+				return fmt.Errorf("dynamodb error: failed to marshal value for key %s: %s", req.Key, err)
+			}
+			twi.Put = &dynamodb.Put{
+				TableName: aws.String(d.table),
+				Item: map[string]*dynamodb.AttributeValue{
+					d.partitionKey: {
+						S: aws.String(req.Key),
+					},
+					"value": {
+						S: aws.String(value),
+					},
+				},
+			}
+
+		case state.DeleteRequest:
+			twi.Delete = &dynamodb.Delete{
+				TableName: aws.String(d.table),
+				Key: map[string]*dynamodb.AttributeValue{
+					d.partitionKey: {
+						S: aws.String(req.Key),
+					},
+				},
+			}
+		}
```

Comment:
Do we need a default statement here passing back unsupported operation?

Reply:
I don't know if it will provide any value currently as `Set` and `Delete` are the only possible operations. If in the future, another operation is added, IMO it should be driven by the conformance tests.